File: samples/elasticsearch-java/app/controllers/Application.java
package controllers;

import com.github.cleverage.elasticsearch.IndexQuery;
import com.github.cleverage.elasticsearch.IndexResults;
import com.github.cleverage.elasticsearch.IndexService;
import indexing.IndexTest;
import indexing.Team;
import org.elasticsearch.index.query.QueryBuilders;
import org.elasticsearch.search.aggregations.AggregationBuilders;
import org.elasticsearch.search.aggregations.bucket.terms.TermsAggregator;
import play.libs.F;
import play.mvc.Controller;
import play.mvc.Result;
import views.html.index;

import java.util.List;

public class Application extends Controller {

  public static Result index() {

    // ElasticSearch HelloWorld
    IndexTest indexTest = new IndexTest();
    // "id" is mandatory if you want to update your document or to get by id else "id" is not mandatory
    indexTest.id = "1";
    indexTest.name = "hello World";
    indexTest.index();

    IndexTest byId = IndexTest.find.byId("1");

    IndexResults<IndexTest> all = IndexTest.find.all();

    IndexQuery<IndexTest> indexQuery = IndexTest.find.query();
    indexQuery.setBuilder(QueryBuilders.queryStringQuery("hello"));

    IndexResults<IndexTest> indexResults = IndexTest.find.search(indexQuery);


    // Team indexing
    // search All
    IndexResults<Team> allTeam = Team.find.all();

    // search All + aggregation country

    IndexQuery<Team> queryCountry = Team.find.query();
    queryCountry.addAggregation(AggregationBuilders.terms("countryF").field("country.name"));
    IndexResults<Team> allAndAggregationCountry = Team.find.search(queryCountry);
    TermsAggregator countryF = allAndAggregationCountry.aggregations.get("countryF");

    // search All + aggregation players.position
    IndexQuery<Team> queryPlayers = Team.find.query();
    queryPlayers.addAggregation(AggregationBuilders.terms("playersF").field("players.position"));
    queryPlayers.addAggregation(AggregationBuilders.nested("players"));
    IndexResults<Team> allAndAggregationAge = Team.find.search(queryPlayers);


    return ok(index.render("Your new application is ready."));
  }

    public static F.Promise<Result> async() {
        // ElasticSearch HelloWorld
        IndexTest indexTest = new IndexTest();
        // "id" is mandatory if you want to update your document or to get by id else "id" is not mandatory
        indexTest.id = "1";
        indexTest.name = "hello World";
        indexTest.index();

        // ElasticSearch HelloWorld
        IndexTest indexTest2 = new IndexTest();
        // "id" is mandatory if you want to update your document or to get by id else "id" is not mandatory
        indexTest.id = "2";
        indexTest.name = "hello Bob";
        indexTest.index();

        IndexService.refresh();

        IndexQuery<IndexTest> query1 = IndexTest.find.query();
        query1.setBuilder(QueryBuilders.matchQuery("name", "hello"));
        IndexQuery<IndexTest> query2 = IndexTest.find.query();
        query2.setBuilder(QueryBuilders.matchQuery("name", "bob"));

        F.Promise<IndexResults<IndexTest>> indexResultsPromise1 = IndexTest.find.searchAsync(query1);
        F.Promise<IndexResults<IndexTest>> indexResultsPromise2 = IndexTest.find.searchAsync(query2);

        F.Promise<List<IndexResults<IndexTest>>> combinedPromise = F.Promise.sequence(indexResultsPromise1, indexResultsPromise2);
        return combinedPromise.map(new F.Function<List<IndexResults<IndexTest>>, Result>() {
                @Override
                public Result apply(List<IndexResults<IndexTest>> indexResultsList) throws Throwable {
                    return ok(indexResultsList.get(0).totalCount + " - " + indexResultsList.get(1).totalCount);
                }
            }
        );

    }

}

File: samples/elasticsearch-scala/app/controllers/Application.scala
package controllers

import com.github.cleverage.elasticsearch.ScalaHelpers.{IndexQuery, _}
import com.github.cleverage.elasticsearch.component.IndexComponent
import javax.inject.{Singleton, Inject}
import indexing.{IndexTest, IndexTestManager}
import org.elasticsearch.index.query.QueryBuilders
import play.api._
import play.api.libs.concurrent.Execution.Implicits._
import play.api.mvc._

import scala.concurrent.Future

@Singleton
class Application @Inject() (indexComponent: IndexComponent) extends Controller {
  
  def index = Action {
    val indexTest = IndexTest("1", "The name", "The category")
    IndexTestManager.index(indexTest)
    Logger.info("IndexTestManager.index() : " + indexTest)

    val gettingIndexTest = IndexTestManager.get("1")
    Logger.info("IndexTestManager.get() => " + gettingIndexTest)

    IndexTestManager.delete("1")
    Logger.info("IndexTestManager.delete()");

    val gettingIndexTestMore = IndexTestManager.get("1")
    Logger.info("IndexTestManager.get() => " + gettingIndexTestMore)

    IndexTestManager.index(IndexTest("1", "Here is the first name", "First category"))
    IndexTestManager.index(IndexTest("2", "Then comes the second name", "First category"))
    IndexTestManager.index(IndexTest("3", "Here is the third name", "Second category"))
    IndexTestManager.index(IndexTest("4", "Finnaly is the fourth name", "Second category"))

    val indexQuery = IndexQuery[IndexTest]()
      .withBuilder(QueryBuilders.matchQuery("name", "Here"))
    val results: IndexResults[IndexTest] = IndexTestManager.search(indexQuery)

    Logger.info("IndexTestManager.search()" + results);
    //IndexTestManager.delete("1")
    //IndexTestManager.delete("2")
    //IndexTestManager.delete("3")
    //IndexTestManager.delete("4")

    Ok(views.html.index("Your new application is ready."))
  }

  def async = Action.async {
    IndexTestManager.index(IndexTest("1", "Here is the first name", "First category"))
    IndexTestManager.index(IndexTest("2", "Then comes the second name", "First category"))
    IndexTestManager.index(IndexTest("3", "Here is the third name", "Second category"))
    IndexTestManager.index(IndexTest("4", "Finnaly is the fourth name", "Second category"))

    IndexTestManager.refresh()

    val indexQuery = IndexTestManager.query
      .withBuilder(QueryBuilders.matchQuery("name", "Here"))
    val indexQuery2 = IndexTestManager.query
      .withBuilder(QueryBuilders.matchQuery("name", "third"))

    // Combining futures
    val l: Future[(IndexResults[IndexTest], IndexResults[IndexTest])] = for {
      result1 <- IndexTestManager.searchAsync(indexQuery)
      result2 <- IndexTestManager.searchAsync(indexQuery2)
    } yield (result1, result2)

    l.map { case (r1, r2) =>
      Ok(r1.totalCount + " - " + r2.totalCount)
    }

  }
  
}